### ISSUE TITLE
Calling detectMimeType(file) with zero length file will forever block thread

### ISSUE THREAD

**Issue by Enerccio** [overview/mime-types#2]
and it can even deadlock whole webapp. This happens when file requested is empty. 

Helper thread will fail with 
```
Exception in thread "Thread-6" java.lang.NegativeArraySizeException

at org.overviewproject.mime_types.MimeTypeDetector$2.completed(MimeTypeDetector.java:297)
	at org.overviewproject.mime_types.MimeTypeDetector$2.completed(MimeTypeDetector.java:295)
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)
	at sun.nio.ch.SimpleAsynchronousFileChannelImpl$2.run(SimpleAsynchronousFileChannelImpl.java:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```
but this will forever deadlock the original thread that was asking for mime type information since future will never end. This should crash more gracefully than that. Sure, I can add checks before calling detectMimeType but this should not deadlock the thread.

**Comment by adamhooper** [comment]
@Enerccio Thank you for the issue report! I've just released v0.1.1 with a fix. It should be available in the Maven central repository within an hour or two.

**Comment by Enerccio** [comment]
Thanks for the fix!
